nautobot/extras/groups.py

```diff
@@ -0,0 +1,201 @@
+from collections import OrderedDict
+import logging
+import urllib
+
+import django_filters
+from django_filters.utils import get_model_field
+from django.db.models import Q
+from django.db import models
+from django.urls import reverse
+from django.utils.functional import classproperty
+
+from nautobot.utilities.utils import get_filterform_for_model, get_filterset_for_model, get_route_for_model
+
+
+logger = logging.getLogger(__name__)
+
+
+def extract_value_from_object_from_queryset(obj, field_parts):
```

Comment:
From @glennmatthews:

> Need a docstring and example usage here.

Also it's not obvious to me the significance of naming this as _from_queryset here - seems like this is just extracting a (potentially nested) attribute from an object?